lib/events.js

```diff
@@ -385,8 +386,12 @@ EventEmitter.prototype.removeListener =
           this._events = Object.create(null);
         else {
           delete events[type];
-          if (events.removeListener)
-            this.emit('removeListener', type, list.listener || listener);
+          if (events.removeListener) {
+            this.emit('removeListener',
+                      type,
+                      list.listener || listener,
```

Comment:
I wonder for this case and the one above if it would affect benchmarks in a positive way to do `const onceListener = list.listener` and then reuse that in both arguments to avoid the duplicate property lookup. It may not register, but it would be interesting to see nonetheless.

Reply:
I've tried your suggestion and these were the results (being *old* without your suggestion and *new* with it):
```
events/ee-add-remove.js n=250000                        -5.02 %        *** 5.764014e-07
events/ee-emit.js listeners=1 argc=0 n=2000000           0.65 %            1.645928e-01
events/ee-emit.js listeners=1 argc=10 n=2000000          0.32 %            3.825945e-01
events/ee-emit.js listeners=1 argc=2 n=2000000           0.64 %            1.429730e-01
events/ee-emit.js listeners=1 argc=4 n=2000000           0.49 %            3.639564e-01
events/ee-emit.js listeners=10 argc=0 n=2000000         -0.20 %            5.467603e-01
events/ee-emit.js listeners=10 argc=10 n=2000000        -0.00 %            9.975995e-01
events/ee-emit.js listeners=10 argc=2 n=2000000         -0.04 %            9.082705e-01
events/ee-emit.js listeners=10 argc=4 n=2000000         -0.13 %            6.573266e-01
events/ee-emit.js listeners=5 argc=0 n=2000000           0.61 %            2.240547e-01
events/ee-emit.js listeners=5 argc=10 n=2000000         -0.05 %            9.056139e-01
events/ee-emit.js listeners=5 argc=2 n=2000000           0.13 %            7.647748e-01
events/ee-emit.js listeners=5 argc=4 n=2000000           0.54 %          * 4.065729e-02
events/ee-listener-count-on-prototype.js n=50000000      0.25 %            5.186799e-01
events/ee-listeners-many.js n=5000000                   -0.91 %          * 1.377654e-02
events/ee-listeners.js n=5000000                         0.22 %            6.412473e-01
events/ee-once.js n=20000000                             0.47 %            5.795604e-01
```